copy/src/App.tsx

```diff
@@ -0,0 +1,124 @@
+import { AppState } from "./redux/reducers";
+
+import {
+  setCurrentNumber,
+  setGameOver,
+  setLoading,
+  setQuestions,
+  setScore,
+  setUserAnswers,
+} from "./redux/actions";
+
+export type AnswerObject = {
+  question: string;
+  userAnswer: string;
+  isCorrect: boolean;
+  correctAnswer: string;
+};
+
+const TOTAL_QUESTIONS = 10;
+
+function App() {
+  const isLoading = useSelector((state: AppState) => state.game.loading);
+  const questions = useSelector((state: AppState) => state.game.questions);
+  const userAnswers = useSelector((state: AppState) => state.game.userAnswers);
+  const score = useSelector((state: AppState) => state.game.score);
+  const currentNumber = useSelector(
+    (state: AppState) => state.game.currentNumber
+  );
+  const isGameOver = useSelector((state: AppState) => state.game.gameOver);
+  const dispatch = useDispatch();
+
+  const startQuiz = async () => {
+    dispatch(setLoading(true));
+    dispatch(setGameOver(false));
+
+    const newQuestions = await fetchQuizQuestions(
+      TOTAL_QUESTIONS,
+      Difficulty.Easy
+    );
+
+    dispatch(setQuestions(newQuestions));
+    dispatch(setScore(0));
+    dispatch(setUserAnswers([]));
+    dispatch(setCurrentNumber(0));
+    dispatch(setLoading(false));
+  };
+
+  const checkAnswer = (e: React.MouseEvent<HTMLButtonElement>) => {
+    if (!isGameOver) {
+      e.preventDefault();
+      //user answer
+      const answer = e.currentTarget.value;
+      const isCorrect = questions[currentNumber].correct_answer === answer;
+      if (isCorrect) {
+        dispatch(setScore(score + 1));
+      }
+      const answerObject = {
+        question: questions[currentNumber].question,
+        userAnswer: answer,
+        isCorrect: isCorrect,
+        correctAnswer: questions[currentNumber].correct_answer,
+      };
+      dispatch(setUserAnswers([...userAnswers, answerObject]));
+    }
+  };
+
+  const nextQuestion = () => {
+    dispatch(setCurrentNumber(currentNumber + 1));
+    if (currentNumber + 1 === TOTAL_QUESTIONS) {
+      dispatch(setGameOver(true));
+    } else {
+    }
+  };
+
+  return (
+    <>
+      <GlobalStyle />
+      <Wrapper>
+        <h1>REACT QUIZ</h1>
+
+        {isGameOver || userAnswers.length === TOTAL_QUESTIONS ? (
+          <button className="start" onClick={startQuiz}>
+            Start
+          </button>
+        ) : null}
+
+        {!isGameOver ? <p className="score">Score: {score}</p> : null}
+        {isLoading ? (
+        <p>Loading...</p>
+        ) : null}
+        {!isLoading && !isGameOver ? (
+          <QuestionCard
+            questionNum={currentNumber + 1}
+            totalQuestions={TOTAL_QUESTIONS}
+            question={questions[currentNumber].question}
+            answers={questions[currentNumber].answers}
+            userAnswer={userAnswers ? userAnswers[currentNumber] : undefined}
+            callback={checkAnswer}
+          />
+        ) : null}
+        {!isGameOver &&
```

Comment:
since !isGameOver && !isLoading condition repeating, you should maybe nest this condition within the above one.

If you want to add two react comps under one condition, you can wrap them with a fragment, which in real html will not generate anything, eg.
`<><OneComponent/><SecondComponent/></>`